### PR TITLE
Enable typescript validation with eslint and update CI build/test scripts

### PR DESCRIPTION
**Pull request by jerolimov**
Changes are similar to https://github.com/RedHatInsights/hac-core-frontend/pull/13

* Removed unused babel dependencies
* Replaced `.eslintrc.yml` with `.eslintrc.js` from hac-core
* Fix eslint issues
* Updated `build.sh` and `test.sh` to run `npm run build` and `npm run verify` on CI

### CODE REVIEW

**Comment by andrewballantyne** [comment]
Appears to be an issue with your lint setup. I see the same thing locally. 

FWIW, I setup our hac-core-frontend like:

- [.eslintrc.js](https://github.com/RedHatInsights/hac-core-frontend/blob/master/.eslintrc.js)
- [npm run script for lint](https://github.com/RedHatInsights/hac-core-frontend/blob/master/package.json#L18)

File: package.json [hunk 1]
```diff
@@ -29,7 +29,10 @@
     "@redhat-cloud-services/frontend-components": "^3.3.3",
     "@redhat-cloud-services/frontend-components-notifications": "^3.2.2",
     "@redhat-cloud-services/frontend-components-utilities": "^3.2.4",
+    "@typescript-eslint/eslint-plugin": "^5.3.0",
     "classnames": "^2.3.1",
+    "eslint-plugin-import": "^2.25.2",
+    "eslint-plugin-react-hooks": "^4.2.0",
```
Comment (andrewballantyne) [inline package.json:35]: Might want to move these to the devDeps section.

Comment (jerolimov) [reply package.json]: Thanks, fixed. :+1:

**Review by andrewballantyne — APPROVED** [review]

**Comment by andrewballantyne** [comment]
Happy coding HAC-Dev folks 😄 

Let me know if there are any other issues with the CI. I posted in your slack channel as well.

**Comment by andrewballantyne** [comment]
laaaaaaaaame... meant to wait for Rohit

**Comment by jerolimov** [comment]
Too late. Damn... Who can expect such a quick CI... :smirk:

**Pull request merged**